ecscollector/collector.go

```diff
@@ -223,41 +275,50 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 			continue
 		}
 
-		labelVals := []string{
+		containerLabelVals := []string{
 			container.Name,
+			container.Image,
```

Comment:
Do we need the container image name to identify the uniqueness? I think this would be better included in an "info" metric.

Reply:
The concern is vaguely that someone can use the same container name in different task definitions differently, making it a misleading dimension for aggregation.  But the same can be said for an image name.  You could only be certain about this if you instead used labels that identify the task definition (family+revision).

Anyway, curious what your thoughts are on this.

Reply:
Identification of different instances of the same container is up to the service discovery.

The main thing we need to make sure is that have a fully unique key or keys as we `range metadata.Containers`.

Reply:
> Identification of different instances of the same container is up to the service discovery.

It makes sense that service discovery is responsible for generating distinct label sets for distinct instances, but is it also solely responsible for generating label sets that are useful for doing useful aggregations across the resulting timeseries?

If I want to, for example, compare container memory usage across two versions of an ECS service during a deployment (i.e. the current and previous version of the service), one way to distinguish between those versions would be with this `image` label, or by a combination of the ECS task family and revision labels.  `container_name` alone can’t help me do this, unless I was doing something strange like changing the name in each task definition revision.

Are you saying we should always leave the addition of these other potentially useful labels to service discovery? My concern is that it might be much harder for service discovery to generate them than it is for us, here in the exporter. If you’re using Cloud Map DNS service discovery, all you’re going to get is an IP address. You’d have to implement http_sd using a Cloud Map API client to have a chance to do this kind of thing, I think.

Reply:
> Are you saying we should always leave the addition of these other potentially useful labels to service discovery?

No, I'm saying that we should probably follow the "info metric" pattern. 

For example, in node_exporter we have `node_network_info`, which has a number of labels for things that are useful, but not really necessary all the time on every other `node_network_...` metric.

There are cases where we do include some "not required for uniqueness" labels on all related metrics. The `fstype` label on `node_filesystem_...` is a good example of this. It's not required for uniqueness, but it's very common to filter by `fstype`. Much more common than say, all of the other mountpoint flags.

This is one of those "naming is hard" problems. I just want to make sure we think about why we're including labels.

Reply:
How can you use such info metrics though? How do you take labels on one metric and apply them to another? Prometheus doesn’t support join-type operations across timeseries as far as I know. Can you achieve something like that during relabeling, or?

Reply:
> Prometheus doesn’t support join-type operations across timeseries as far as I know

It very much does support this. Here's a [blog post on the subject](https://grafana.com/blog/2021/08/04/how-to-use-promql-joins-for-more-effective-queries-of-prometheus-metrics-at-scale/).

There's also a new, experimental, `info()` function being worked on.

Reply:
Welp, thanks for that link.  Alright then, I'll revert these label additions.

Under this reasoning, I think the task-level timeseries should in fact have no labels, when they currently have `task_arn`.  I'd like to take the opportunity to drop it now.  Sound good?

Reply:
Yes, that makes sense. We can always add them back later if we find that they're necessary.

Reply:
Great, thanks for handholding through this, I will do another round of testing and then update the README as per my above issue.